overlord/devicestate/devicectx.go

```diff
@@ -30,11 +30,14 @@ func DeviceCtx(st *state.State, task *state.Task, providedDeviceCtx snapstate.De
 	if providedDeviceCtx != nil {
 		return providedDeviceCtx, nil
 	}
-	modelAs, err := findModel(st)
+	if remodelCtx, err := remodelCtxFromTask(task); err == nil {
```

Comment:
we should distinguish ErrNoState from other errors, also this needs to be covered by unit tests

Reply:
Code for this was added in ce42a2882c93f4b599888ba086b19bf18deef564